sklearn/utils/validation.py

```diff
@@ -980,3 +980,56 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def _check_sample_weight(sample_weight, X, dtype=None,
+                         order=None):
+    """Validate sample weights
+
+    Parameters
+    ----------
+    sample_weight : {ndarray, Number or None}, shape (n_samples,)
+       Input sample weights.
+    X : nd-array, list or sparse matrix
+        Input data.
+    dtype: dtype
+       dtype of the validated `sample_weight`. Note that if `dtype` is not
+       one of `float32`, `float64`, the output will be of dtype `float64`.
+    order : 'F', 'C' or None (default=None)
+        Whether an array will be forced to be fortran or c-style.
```

Comment:
'F' is equivalent to 'C' for 1D array. Don't you think we can drop this parameter and always enforce 'C'.
Letting non contiguous array pass is risky if it's passed to cython, with pointers manipulation

Reply:
I agree.